load_cart.php

```diff
@@ -1,64 +1,86 @@
 <?php
 session_start();
-include 'connectdb.php';  // Reuse existing database connection
+
+$userType = $_SESSION['type'];
 
 if (!isset($_SESSION['order_id'])) {
     die('Order ID not set in session.');
 }
-
 $order_id = $_SESSION['order_id'];
 
-try {
-    // Fetch order details with dish information in a single query
-    $sql = "SELECT od.*, m.dish_name, m.image 
-            FROM order_detail od 
-            JOIN menu m ON od.dish_id = m.dish_id 
-            WHERE od.order_id = :order_id AND od.status = 'inprogress'";
+// Database connection
+$conn = new mysqli('localhost', 'root', '', 'rms');
+if ($conn->connect_error) {
+    die("Connection failed: " . $conn->connect_error);
+}
+
+// Fetch order details
+$sql = "SELECT * FROM order_detail WHERE order_id = ? AND status = 'inprogress'";
+$stmt = $conn->prepare($sql);
+$stmt->bind_param("i", $order_id);
+$stmt->execute();
+$result = $stmt->get_result();
+$orderDetails = $result->fetch_all(MYSQLI_ASSOC);
+$stmt->close();
 
+// Fetch dish names and images
+foreach ($orderDetails as &$detail) {
+    $dish_id = $detail['dish_id'];
+    $sql = "SELECT dish_name, image FROM menu WHERE dish_id = ?";
     $stmt = $conn->prepare($sql);
-    $stmt->bindParam(':order_id', $order_id, PDO::PARAM_INT);
+    $stmt->bind_param("i", $dish_id);
```

Comment:
<div>


<div id="suggestion">
<div id="issue"><b>Consider consistent database API usage</b></div>
<div id="fix">

Consider using `bindParam()` instead of `bind_param()` for consistency since the code uses PDO elsewhere. The current mix of mysqli and PDO could lead to maintenance issues.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<blockquote>Check the AI-generated fix before applying</blockquote>
<div id="code">


```suggestion
    $stmt->bindParam(':dish_id', $dish_id, PDO::PARAM_INT);
```

</div>
</details>
</div>


<small><i>Code Review Run <a href=https://github.com/Lux-Jason/rms_new/pull/12#issuecomment-2561647433>#75a1e4</a></i></small>
</div>

---
Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged